proposals/public-suffix.md

```diff
@@ -0,0 +1,555 @@
+
+##### Example
+
+`domain` = foo.bar.wixsite.com
+
+| Option                                   |   Registrable Domain | PSL Section |
+|------------------------------------------|:--------------------:|:-----------:|
+| excludePrivateSuffixes = false (default) |      bar.wixsite.com | Private     |
+| excludePrivateSuffixes = true            |          wixsite.com | ICANN       |
+
+#### 2. Multiple Suffixes per Domain
+
+The lookup performed by `getRegistrableDomain()` should select the **longest** matching
+suffix (unless specifically excluded using the `excludePrivateSuffixes` option).
+
+##### Example
+
+`domain` = foo.bar.lib.de.us
+
+| Candidate Suffix | PSL Section |
+|-----------------:|:-----------:|
+|            de.us | ICANN       |
+|        lib.de.us | Private     |
+
+The longest is lib.de.us, so `getRegistrableDomain()` resolves to bar.lib.de.us
+
+#### 3. PSL Special Rules
+
+The lookup performed by `getRegistrableDomain()` should adhere to the
+[PSL algorithm](https://github.com/publicsuffix/list/wiki/Format#formal-algorithm).
+In particular, it should apply the 'wildcard' and 'exception' rules in the PSL.
+
+##### Examples
+
+| Domain                   | Public suffix | Matched PSL rule | Explanation    | Registrable Domain |
+|-------------------------:|--------------:|-----------------:|:--------------:|-------------------:|
+| sub.domain.com           | com           | com              | Simple rule    | domain.com         |
+| sub.domain.co.uk         | co.uk         | co.uk            | Simple rule    | domain.co.uk       |
+| sub.domain.gov.ck        | gov.ck        | *.ck             | Wildcard rule  | domain.gov.ck      |
+| sub.domain.any.ck        | any.ck        | *.ck             | Wildcard rule  | domain.any.ck      |
+| sub.sub.domain.any.ck    | any.ck        | *.ck             | Wildcard rule  | domain.any.ck      |
+| www.ck                   | ck            | !www.ck          | Exception rule | www.ck             |
+| sub.www.ck               | ck            | !www.ck          | Exception rule | www.ck             |
+| sub.sub.www.ck           | ck            | !www.ck          | Exception rule | www.ck             |
+
+#### 4. Unknown Suffixes
+
+If no matching suffix is found in the PSL for a `domain` parameter, then unless it is determined
+to be specifically [invalid](#6-invalid-domain-parameter), it should be assumed the domain has a
+single-label suffix.
+
+##### Example
+
+| Domain parameter       | Registrable domain | PSL Section |
+|-----------------------:|-------------------:|:-----------:|
+| www.example.foobar     | example.foobar     | n/a         |
+| www.example.co.foobar  | co.foobar          | n/a         |
+
+#### 5. IDN
+
+The `domain` parameter passed to `getRegistrableDomain()` may be either Unicode
+or Punycode.
+
+When settling the promise returned by `getRegistrableDomain()`, the resulting
+domain name should be converted to Unicode from Punycode by default.
```

Comment:
Why should the API convert the domain to Unicode by default? When the input is the host name from a URL, it would be reasonable to expect a valid hostname. I'd argue that punycode is the more sane default.

Extensions can bundle the custom library/logic to convert punycode to unicode if desired, because the algorithm is well known and fixed.

Reply:
I can change this to returning punycode by default, with an API option to convert to unicode. Unicode registrable domains are required by Use Case 2 (Group Domains in UI) in this proposal.